### DISCUSSION TITLE
How to deploy with gradle new plugin

### DISCUSSION THREAD

**Discussion by zepouet** [allure-framework/allure2#1767] [Questions & Support] [answered]
Hello

Sorry if the question is very easy but for me not. Could you explain which Gradle tasks to run to deploy automatically an existing example plugin or a new one ?

I have idea to contribute about differents plugins but no way to enable them with gradle.
I know very well Maven but never use gradle.

In all case, I could help to contribute to a quick start dev tutorial after I understand.

Thanks a lot

**Comment by Avaddon-tfa** [comment]
Hey there, thanks for the question. It's all hands on deck now hence we stay silent. 

I am passing your request right to the Report devs, so I hope the answer will come soon.

**Comment by baev** [comment, marked as answer]
We do not provide plugins distribution system yet, so when custom plugin is used you need to maintain your own allure-commandline distribution. To do so:

1. Fork allure2 repository
2. Use `./gradlew build` to build the project. We only run build on Java 8 and 11, so in case of any issues with more modern JDK try to downgrade to 11 at first. 
3. Builded distributions location is `allure-commandline/build/distributions`. Unpacked version is also available `allure-commandline/build/install/allure-commandline` (you can create allure-dev symlink to that path)
4. To create a plugin you can use `plugins/behaviors-plugin` as a template.
    1. Copy the module 
    2. Add it to `settings.gradle.kts`
    3. Add plugin as a dependency to allure-commandline module https://github.com/allure-framework/allure2/blob/master/allure-commandline/build.gradle.kts#L128
    4. Add plugin to default profile to enable it by default [allure.yml](https://github.com/allure-framework/allure2/blob/master/allure-commandline/src/dist/config/allure.yml)
    5. Now simply run `./gradlew build` to rebuild the distribution with a new plugin included to it

Guide how to write a plugin could be found here https://docs.qameta.io/allure#_allure_plugins_system

How to dev a plugin:

1. For dev mode we have `./gradlew dev` task. After that webpack dev server will be started at port 3000
2. Dev report generated from [test data (allure results)](https://github.com/allure-framework/allure2/tree/master/allure-generator/test-data/new-demo)
3. You can change the test data here https://github.com/allure-framework/allure2/blob/master/allure-generator/build.gradle.kts#L74
4. In order to enable plugin in dev mode, add it as a dependency to `allure-generator` module https://github.com/allure-framework/allure2/blob/master/allure-generator/build.gradle.kts#L102
